jscomp/others/node_buffer.ml

```diff
@@ -22,21 +22,521 @@
+    @since Node.js v0.1.101
+*)
+external isBuffer : 'a -> bool = "isBuffer" [@@bs.val] [@@bs.scope "Buffer"]
+
+(** Returns [true] if [encoding] contains a supported character encoding, or [false] otherwise.
+
+    @see <https://nodejs.org/dist/latest-v13.x/docs/api/buffer.html#buffer_class_method_buffer_isencoding_encoding> Node.js API docs
+    @since v0.9.1
+  *)
+external isEncoding : string -> bool = "isEncoding" [@@bs.val] [@@bs.scope "Buffer"]
+
+(** This is the size (in bytes) of pre-allocated internal [Buffer] instances
+    used for pooling.
+  
+    @see <https://nodejs.org/dist/latest-v13.x/docs/api/buffer.html#buffer_class_property_buffer_poolsize> Node.js API docs
+    @since Node.js v0.11.3
+  *)
+external poolSize : int = "poolSize" [@@bs.val] [@@bs.scope "Buffer"]
+
+(** Unsafe byte reading by its [index].
+    Underneath it uses index access syntax in Javascript.
+
+    @see <https://nodejs.org/dist/latest-v13.x/docs/api/buffer.html#buffer_buf_index> Node.js API docs
+  *)
+external unsafe_get : t -> int -> int = "" [@@bs.get_index]
+
+(** Unsafe byte writing by its [index].
+    Underneath it uses index access syntax in Javascript.
+
+    @see <https://nodejs.org/dist/latest-v13.x/docs/api/buffer.html#buffer_buf_index> Node.js API docs
+  *)
+external unsafe_set : t -> index:int -> int -> unit = "" [@@bs.set_index]
+
+(** Returns the underlying ArrayBuffer object.
+
+    @see <https://nodejs.org/dist/latest-v13.x/docs/api/buffer.html#buffer_buf_buffer> Node.js API docs
+  *)
+external buffer : t -> Js_typed_array2.ArrayBuffer.t = "buffer" [@@bs.get]
+
+(** Returns [byteOffset] on the underlying [ArrayBuffer] object.
+
+    @see <https://nodejs.org/dist/latest-v13.x/docs/api/buffer.html#buffer_buf_byteoffset> Node.js API docs
+  *)
+external byteOffset : t -> int = "byteOffset" [@@bs.get]
+
+(** Copies data from a region of [buf] to a region in [target].
+
+    @see <https://nodejs.org/dist/latest-v13.x/docs/api/buffer.html#buffer_buf_copy_target_targetstart_sourcestart_sourceend> Node.js API docs
+    @since Node.js v0.1.90
+  *)
+external copy :
+  t ->
+  target:t ->
+  ?targetStart:int ->
+  ?sourceStart:int ->
+  ?sourceEnd:int ->
+  unit -> int = "copy"
+[@@bs.send]
+
+(** Returns [true] if both [buf] and [otherBuffer] have exactly the same bytes.
+
+    @see <https://nodejs.org/dist/latest-v13.x/docs/api/buffer.html#buffer_buf_equals_otherbuffer> Node.js API docs
+    @since Node.js v0.11.13
+  *)
+external equals : t -> t -> bool = "equals" [@@bs.send]
+
+(** Fills [buf] with the specified [value].
+
+    @see <https://nodejs.org/dist/latest-v13.x/docs/api/buffer.html#buffer_buf_fill_value_offset_end_encoding> Node.js API docs
+    @since Node.js v0.5.0
+ *)
+external fill :
+  t ->
+  ([ `Integer of int | `Buffer of t | `Uint8Array of Js_typed_array2.Uint8Array.t ] [@bs.unwrap]) ->
+  ?offset:int ->
```

Comment:
cc @cristianoc 
as you can see, compile  polyvar as string would save lots of  use cases of bs.unwrap/bs.string which is  kinda of a hack here

Reply:
@bobzhang indeed! Let's tackle that next, together with a general look at ordinary+poly variants.